client/wfm/src/service/DataService.ts

```diff
@@ -0,0 +1,8 @@
+import * as Promise from 'bluebird';
+export interface DataService<T> {
```

Comment:
Yes, this would be the interface the user needs to implement to have another, non-sync-based storage.

Just the result service right now needs an extra method, and that's reflected in the constructor for the wfm service.